crates/context/config/src/client/env/config/types/evm.rs

```diff
@@ -0,0 +1,289 @@
+                .iter()
+                .map(|m| FixedBytes::from(m.as_bytes()))
+                .collect();
+
+            // Encode the members array
+            SolValue::abi_encode(&sol_members)
+        }
+        ContextRequestKind::Grant { capabilities } => {
+            // For Grant, we need to encode the list of (member, capability) pairs
+            // This depends on how your contract expects the data
+
+            // If your contract expects an array of members and an array of capabilities:
+            let members: Vec<FixedBytes<32>> = capabilities
+                .iter()
+                .map(|(member, _)| FixedBytes::from(member.as_bytes()))
+                .collect();
+
+            let capability_values: Vec<u8> = capabilities
+                .iter()
+                .map(|(_, capability)| capability.to_sol() as u8)
+                .collect();
+
+            // Encode as (bytes32[] members, uint8[] capabilities)
+            // This requires manual ABI encoding
+
+            // First, encode each array separately
+            let encoded_members = SolValue::abi_encode(&members);
+            let encoded_capabilities = SolValue::abi_encode(&capability_values);
+
+            // Now construct the tuple encoding
+            let mut data = Vec::new();
+
+            // Add offset to first array (32 bytes)
+            let mut offset1_bytes = [0u8; 32];
+            offset1_bytes[31] = 64; // Offset is 64 bytes (2 * 32)
+            data.extend_from_slice(&offset1_bytes);
+
+            // Add offset to second array (32 bytes)
+            let mut offset2_bytes = [0u8; 32];
+            // The offset to the second array is 64 (for the two offsets) plus the length of the first array
+            let offset2 = 64 + encoded_members.len();
+            // We need to encode this as a 32-byte big-endian integer
+            for i in 0..4 {
+                // Assuming the length won't exceed 2^32
+                offset2_bytes[28 + i] = ((offset2 >> (8 * (3 - i))) & 0xFF) as u8;
+            }
+            data.extend_from_slice(&offset2_bytes);
+
+            // Add encoded arrays
+            data.extend_from_slice(&encoded_members);
+            data.extend_from_slice(&encoded_capabilities);
+
+            data
+        }
+        ContextRequestKind::Revoke { capabilities } => {
+            // Same encoding as Grant
+            // For Revoke, we need to encode the list of (member, capability) pairs
+            // This depends on how your contract expects the data
+
+            // If your contract expects an array of members and an array of capabilities:
+            let members: Vec<FixedBytes<32>> = capabilities
+                .iter()
+                .map(|(member, _)| FixedBytes::from(member.as_bytes()))
+                .collect();
+
+            let capability_values: Vec<u8> = capabilities
+                .iter()
+                .map(|(_, capability)| capability.to_sol() as u8)
+                .collect();
```

Comment:
[`Iterator::unzip`](https://doc.rust-lang.org/beta/std/iter/trait.Iterator.html#method.unzip) or [`.collect::<(Vec<_>, Vec<_>)>`](https://doc.rust-lang.org/beta/std/iter/trait.FromIterator.html#impl-FromIterator%3C(EA,)%3E-for-(A,))